packages/host/app/components/operator-mode/card-url-bar.gts

```diff
@@ -0,0 +1,178 @@
+      {{#if this.showErrorMessage}}
+        <div class='error-message' data-test-card-url-bar-error>
+          <span>{{this.errorMessage}}</span>
+        </div>
+      {{/if}}
+    </div>
+    <style>
+      .card-url-bar {
+        display: flex;
+        align-items: center;
+
+        background: var(--boxel-purple-700);
+        border-radius: var(--boxel-border-radius-xl);
+        padding: var(--boxel-sp-xs) 0 var(--boxel-sp-xs) var(--boxel-sp-sm);
+
+        width: 100%;
+        position: relative;
+      }
+      .focused {
+        outline: 2px solid var(--boxel-teal);
+      }
+      .error {
+        outline: 2px solid red;
+      }
+      .realm-info {
+        display: flex;
+        align-items: center;
+        gap: var(--boxel-sp-xs);
+
+        width: max-content;
+        color: var(--boxel-light);
+        border-right: 2px solid var(--boxel-purple-300);
+        padding-right: var(--boxel-sp-xs);
+        margin-right: var(--boxel-sp-xs);
+
+        white-space: nowrap;
+      }
+      .realm-info img {
+        width: 22px;
+      }
+      .input {
+        display: flex;
+        align-items: center;
+        gap: var(--boxel-sp-xs);
+        width: 100%;
+
+        --icon-color: var(--boxel-cyan);
+      }
+      .error .input {
+        --icon-color: red;
+      }
+      .url-input {
+        background: none;
+        border: none;
+        border-radius: 0;
+        outline: none;
+        padding: 0;
+        min-height: 0;
+
+        color: var(--boxel-light);
+      }
+      .error-message {
+        position: absolute;
+        bottom: calc(calc(var(--boxel-sp-xs) * 2) * -1);
+        color: red;
+      }
+    </style>
+  </template>
+
+  @service declare cardService: CardService;
+  @tracked url: string = this.args.url.toString();
+  @tracked isFocused = false;
+  @tracked isInvalidURL = false;
+
+  get realmIcon() {
+    return this.fetchRealmInfo.value?.iconURL;
+  }
+
+  get realmName() {
+    return this.fetchRealmInfo.value?.name;
+  }
+
+  get showErrorMessage() {
+    return this.isInvalidURL || this.args.notFoundError;
+  }
+
+  get errorMessage() {
+    if (this.isInvalidURL) return 'Not a valid URL';
+    else return this.args.notFoundError;
+  }
+
+  get cssClasses() {
+    if (this.showErrorMessage) {
+      return 'card-url-bar error';
+    } else if (this.isFocused) {
+      return 'card-url-bar focused';
+    } else {
+      return 'card-url-bar';
+    }
+  }
+
+  fetchRealmInfo = trackedFunction(this, async () => {
+    if (!this.args.card) return;
+    return this.cardService.getRealmInfo(this.args.card);
+  });
+
+  @action
+  onInput(url: string) {
+    this.url = url;
+    this.isInvalidURL = false;
+    this.args.resetNotFoundError();
+  }
+
+  @action
+  onKeyPress(event: KeyboardEvent) {
+    if (event.key !== 'Enter') return;
+
+    let url;
+    try {
+      url = new URL(this.url);
+    } catch (e) {
+      this.isInvalidURL = true;
+    }
+
+    if (url) this.args.onEnterPressed(url);
```

Comment:
let's not do `if` statements without `{}` it's too easy to introduce bugs if we decide we want to add more statements for the `if`

Reply:
we will need to clear the error state in the container when a non-enter key is pressed. I think you'll need to pass another callback argument into this component that allows us to reset the error state if the user starts typing

Reply:
> we will need to reset the error state in the container when a non-enter key is pressed. I think you'll need to pass another callback argument into this component that allows us to reset the error state if the user starts typing

I did it here https://github.com/cardstack/boxel/pull/571/files#diff-7e7fc64a8985fde5bec07503f08e35be28fdfe2e4622996d55595dbde3f0e7a8R18.